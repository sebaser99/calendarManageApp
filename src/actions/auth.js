import Swal from "sweetalert2"
import { fetchConToken, fetchSinToken } from "../helpers/fetch"
import { types } from "../types/types"

export const startLogin = (email, password) => {
    return async (dispatch)=> {
        const resp = await fetchSinToken('auth', {email, password}, 'POST')
        const body = await resp.json()
        
        if(body.ok){
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime())

            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
        } else {
            Swal.fire('Error', body.msg, 'error')
        }
    }
}

export const startResgister = (name, email, password) => {
    return async (dispatch)=> {
        const resp = await fetchSinToken('auth/new', {email, name, password}, 'POST')
        const body = await resp.json()

        if(body.ok){
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime())
            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
        } else {
            return Swal.fire('Error', body.msg, 'error')
        }
    }
}

export const startChecking = () => {
    return async (dispatch)=> {
        const resp = await fetchConToken('auth/renew')
        const body = await resp.json()

        if(body.ok){
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime())
            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
        } else {
                
                dispatch(checkingFinish())
            
        }
    }
}

export const checkingFinish = () => ({type: types.authCheckingFinish})


export const login = (user) => ({
    type: types.authLogin,
    payload: user
})

export const startLogout = () => {
    return (dispatch) => {
        localStorage.clear()
        dispatch(logout())
    }
}

export const logout = () => ({
    type: types.authLogout
})