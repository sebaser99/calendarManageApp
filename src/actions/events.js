import Swal from "sweetalert2";
import { fetchConToken } from "../helpers/fetch";
import { prepareEvent } from "../helpers/prepareEvent";
import { types } from "../types/types";

export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {
        const {name, uid} = getState().auth
        
        try{

            const res = await fetchConToken('events/new', event, 'POST')
            const body = await res.json();

            if(body.ok){
                event.id = body.event.id
                event.user = {
                    _id: uid,
                    name: name
                }

                dispatch(eventAddNew(event))
            }    

        } catch(error){
            console.log(error)
        }

    }
}

const eventAddNew = (event)=> ({
    type: types.eventsAddNew,
    payload: event
})

export const eventSetActive = (event)=> ({
    type: types.eventsSetActive,
    payload: event
})

export const clearActiveEvent = ()=> ({type: types.eventClearActive})

export const startUpdateEvent = (event) => {
    return async (dispatch) => {
        try {
            const res = await fetchConToken(`events/${event.id}`, event, 'PUT')
            const body = await res.json()

            if(body.ok){
                dispatch(updateEvent(event))
            } else {
                 Swal.fire('Error', body.msg, 'error')
            }  
        } catch(error){
         console.log(error)    
        }
    }
}

const updateEvent = (event) => ({
    type: types.eventUpdated,
    payload: event
})

export const startEventDelete = () => {
    
    return async (dispatch, getState) => {
        const {id} = getState().calendar.activeEvent
        try{
            const res = await fetchConToken(`events/${id}`, {}, 'DELETE')
            const body = await res.json();
            if(body.ok){
                dispatch(deletedEvent())
                
            } else {
                Swal.fire('Error', body.msg, 'error')
            }
            Swal.fire('Borrado ', 'Borrado Exitosamente', 'success')

        } catch(error){
            console.log(error)
            
        }
    }
          
}


const deletedEvent = () =>({type: types.eventDeleted})

export const slotNewEvent = (slotEvent) => ({
    type: types.eventSlotNew,
    payload: slotEvent
})

export const slotEventDelete = () => ({
    type: types.eventSlotDelete
})

export const eventsStartLoaded = () => {
    return async (dispatch) => {
        try{
            const res = await fetchConToken('events/')
            const body = await res.json()
            const eventos = prepareEvent(body.eventos) 
            dispatch(eventsLoaded(eventos))
         
        } catch(error) {
            console.log(error)
        }
    }
}

const eventsLoaded = (events) => ({
    type: types.eventLoaded,
    payload: events

})

export const eventsLogout = () =>(
        {
            type: types.eventLogout
            }
    )
        

