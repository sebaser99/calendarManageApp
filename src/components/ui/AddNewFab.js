import React from 'react'
import { useDispatch } from 'react-redux'
import { clearActiveEvent } from '../../actions/events'
import { uiOpenModal } from '../../actions/ui'

export const AddNewFab = () => {
  const dispatch = useDispatch()

  const handleCreateNewEvent = () => {
    dispatch(clearActiveEvent())
    dispatch(uiOpenModal())
   
  }
  return (
    <button className='btn btn-primary fab'
        onClick={handleCreateNewEvent}
    >
        <i className='fas fa-plus'></i>
    </button>
  )
}
