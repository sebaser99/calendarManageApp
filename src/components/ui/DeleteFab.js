import { useDispatch} from "react-redux"
import { startEventDelete } from "../../actions/events"

export const DeleteFab = () => {
    const dispatch = useDispatch()
    const handleDeleteEvent = () =>{
        dispatch(startEventDelete())
    }

  return (
    <button className='btn btn-danger fab-danger'
        onClick={handleDeleteEvent}
    >
        <i className='fas fa-trash'></i>
        <span>Borrar Evento</span>
    </button>
  )
}
