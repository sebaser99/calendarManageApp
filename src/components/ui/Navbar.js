import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { eventsLogout } from '../../actions/events';

export const Navbar = () => {
  const {name} = useSelector(state => state.auth)
  const dispatch = useDispatch()

  const handleLogout = () => {
    dispatch(startLogout())
    dispatch(eventsLogout())
  }
  return (
    <div className='navbar navbar-dark bg-dark mb-4'>
        <span className='navbar-brand'>
           {name}
        </span>
        <button className='btn btn-outline-danger' onClick={handleLogout}>
        <i className="fa-solid fa-right-from-bracket"></i>
            <span> Salir</span>
        </button>
    </div>
  ) 
};
