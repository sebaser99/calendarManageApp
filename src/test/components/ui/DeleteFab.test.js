import React from 'react';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';

import configurStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { DeleteFab } from '../../../components/ui/DeleteFab';
import { startEventDelete } from '../../../actions/events';

const middlewares = [thunk];
const mockStore = configurStore(middlewares);
const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn()

jest.mock('../../../actions/events', ()=>({
    startEventDelete : jest.fn()
}))  

const wrapper = mount(
    <Provider store={store}>
        <DeleteFab />

    </Provider>
)


describe('pruebas en <DeleteFab />', ()=> {
    test('debe mostrarse de forma correcta', ()=> {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe hacer el llamado a la funciÃ³n startEventDelete', ()=> {
        wrapper.find('.btn').simulate('click')
        expect(startEventDelete).toHaveBeenCalled()
    })
})