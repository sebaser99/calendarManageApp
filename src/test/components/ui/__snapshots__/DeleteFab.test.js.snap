// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <DeleteFab /> debe mostrarse de forma correcta 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
    Borrar Evento
  </span>
</button>
`;
